config/machine_id.go

```diff
@@ -0,0 +1,152 @@
+package config
+
+import (
+	"crypto/rand"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/NordSecurity/nordvpn-linux/internal"
+	"github.com/google/uuid"
+)
+
+type MachineIDGetter interface {
+	GetMachineID() uuid.UUID
+}
+
+type FileReader func(fileName string) ([]byte, error)
+type HostNameReader func() (name string, err error)
+
+type GeneratorFn func() ([]byte, error)
+
+type MachineID struct {
+	hostNameReader HostNameReader
+	fileReader     FileReader
+}
+
+func NewMachineID(fileReader FileReader, hostNameReader HostNameReader) MachineID {
+	return MachineID{
+		hostNameReader: hostNameReader,
+		fileReader:     fileReader,
+	}
+}
+
+func (getter MachineID) GetMachineID() uuid.UUID {
+	id, err := getter.generateID()
+	if err == nil {
+		return id
+	}
+
+	// create a random UUID
+	log.Println(internal.ErrorPrefix, "failed to generate machine ID", err)
+	id, err = uuid.NewRandom()
+	if err == nil {
+		return id
+	}
+
+	// Fallback to manually generating a UUID
+	log.Println(internal.ErrorPrefix, "failed to generate random UUID", err)
+	var fallbackUUID uuid.UUID
+	_, err = rand.Read(fallbackUUID[:])
```

Comment:
What value has `fallbackUUID` variable? as this type is [16]byte - is it initialized with 0 ?